src/operator/controllers/access_annotation/intents.go

```diff
@@ -0,0 +1,104 @@
+package access_annotation
+
+import (
+	"context"
+	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	"github.com/otterize/intents-operator/src/shared/serviceidresolver/serviceidentity"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Intent struct {
+	Client        serviceidentity.ServiceIdentity
+	Server        serviceidentity.ServiceIdentity
+	EventRecorder *injectablerecorder.ObjectEventRecorder
+}
+
+type IntentsByService map[serviceidentity.ServiceIdentity][]Intent
+
+type AnnotationIntents struct {
+	IntentsByServer IntentsByService
+	IntentsByClient IntentsByService
+}
+
+type ServiceIdResolver interface {
+	ResolvePodToServiceIdentity(ctx context.Context, pod *v1.Pod) (serviceidentity.ServiceIdentity, error)
+}
+
+func GetIntentsInCluster(
+	ctx context.Context,
+	k8sClient client.Client,
+	serviceIdResolver ServiceIdResolver,
+	recorder *injectablerecorder.InjectableRecorder,
+) (AnnotationIntents, error) {
+	annotationIntents := AnnotationIntents{
+		IntentsByServer: make(IntentsByService),
+		IntentsByClient: make(IntentsByService),
+	}
+
+	podList, err := getAllAnnotationIntentsServers(ctx, k8sClient)
+	if err != nil {
+		return AnnotationIntents{}, errors.Wrap(err)
+	}
+
+	for _, pod := range podList.Items {
+		if pod.GetDeletionTimestamp() != nil {
+			continue
+		}
+
+		clients, ok, err := ParseAccessAnnotations(&pod)
```

Comment:
Maybe we should send an event to the pod if the parse failed?

Reply:
The error is recorded by the pods reconciler already, so no need to double recored it here, but I add a comment to make it clear